docs/source/json-api/production-setup.rst

```diff
@@ -0,0 +1,300 @@
+
+The logging infrastructure leverages structured logging as implemented by the
+`Logstash Logback Encoder <https://github.com/logstash/logstash-logback-encoder/blob/logstash-logback-encoder-6.3/README.md>`__.
+
+Logged events should carry information about the request being served by the
+*HTTP-JSON API* server. When using a traditional logging target (e.g. standard output
+or rotating files) this information will be part of the log description.
+Using a logging target compatible with the Logstash Logback Encoder allows to have rich
+logs with structured information about the event being logged.
+
+
+
+Metrics
+*******
+
+Enable and configure reporting
+------------------------------
+
+
+To enable metrics and configure reporting, you can use the two following CLI options:
+
+- ``--metrics-reporter``: passing a legal value will enable reporting; the accepted values
+  are as follows:
+
+  - ``console``: prints captured metrics on the standard output
+
+  - ``csv://</path/to/metrics.csv>``: saves the captured metrics in CSV format at the specified location
+
+  - ``graphite://<server_host>[:<server_port>]``: sends captured metrics to a Graphite server. If the port
+    is omitted, the default value ``2003`` will be used.
+
+  - ``prometheus://<server_host>[:<server_port>]``: renders captured metrics
+    on a http endpoint in accordance with the prometheus protocol. If the port
+    is omitted, the default value ``55001`` will be used. The metrics will be
+    available under the address ``http://<server_host>:<server_port>/metrics``.
+
+- ``--metrics-reporting-interval``: metrics are pre-aggregated on the sandbox and sent to
+  the reporter, this option allows the user to set the interval. The formats accepted are based
+  on the ISO-8601 duration format ``PnDTnHnMn.nS`` with days considered to be exactly 24 hours.
+  The default interval is 10 seconds.
+
+Types of metrics
+================
+
+This is a list of type of metrics with all data points recorded for each.
+Use this as a reference when reading the list of metrics.
+
+Counter
+-------
+
+Number of occurrences of some event.
+
+Meter
+-----
+
+A meter tracks the number of times a given event occurred (throughput). The following data
+points are kept and reported by any meter.
+
+- ``<metric.qualified.name>.count``: number of registered data points overall
+- ``<metric.qualified.name>.m1_rate``: number of registered data points per minute
+- ``<metric.qualified.name>.m5_rate``: number of registered data points every 5 minutes
+- ``<metric.qualified.name>.m15_rate``: number of registered data points every 15 minutes
+- ``<metric.qualified.name>.mean_rate``: mean number of registered data points
+
+Timers
+------
+
+A timer records all metrics registered by a meter and by an histogram, where
+the histogram records the time necessary to execute a given operation (unless
+otherwise specified, the precision is nanoseconds and the unit of measurement
+is milliseconds).
```

Comment:
I'm a bit confused by these units. Does that mean we report fractional numbers?

Reply:
So AFAIK codahale library measures it in nanoseconds, but then its down to most of the reporters on how they choose to report it.  (ref:  https://metrics.dropwizard.io/4.2.0/getting-started.html#timers)
e.g the console reporter prints this
```
daml.http_json_api.incoming_json_parsing_and_validation_timing
             count = 1
         mean rate = 0.05 calls/second
     1-minute rate = 0.17 calls/second
     5-minute rate = 0.19 calls/second
    15-minute rate = 0.20 calls/second
               min = 225.41 milliseconds
               max = 225.41 milliseconds
              mean = 225.41 milliseconds
            stddev = 0.00 milliseconds
            median = 225.41 milliseconds
              75% <= 225.41 milliseconds
              95% <= 225.41 milliseconds
              98% <= 225.41 milliseconds
              99% <= 225.41 milliseconds
            99.9% <= 225.41 milliseconds
```

Reply:
Do we care to specify the precision here? Wouldn't

> .. to execute a given operation (in fractional milliseconds).

be good enough here?